packages/css/search.css

```diff
@@ -1,170 +1,108 @@
-  position: absolute;
-  background: none;
-  border: none;
-  right: 0.6em;
-  top: 50%;
-  transform: translateY(-50%);
-  cursor: pointer;
-  height: var(--dsc-search-clear-button-size);
-  width: var(--dsc-search-clear-button-size);
-  border-radius: var(--ds-border-radius-default);
-  font-size: 1.25rem;
-  padding: 0;
-  z-index: 2;
-}
-
-.ds-search--sm .ds-search__input {
-  --dsc-search-clear-button-size: var(--ds-sizing-4);
-
-  height: var(--ds-sizing-10);
-  padding: 0 var(--ds-spacing-3);
-  padding-right: 2.5em;
-}
-
-.ds-search--sm .ds-search__icon {
-  left: var(--ds-spacing-3);
-}
-
-.ds-search--md .ds-search__input {
-  --dsc-search-clear-button-size: var(--ds-sizing-6);
-
-  height: var(--ds-sizing-12);
-  padding: 0 var(--ds-spacing-4);
-  padding-right: 2.2em;
-}
-
-.ds-search--md .ds-search__icon {
-  left: var(--ds-spacing-4);
-}
+  /* The simple variant, when no submit button is present */
+  &:not(:has([type='submit'])) {
+    & input {
+      padding-left: 2.4em;
+    }
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: var(--ds-sizing-2);
+      top: 50%;
+      transform: translateY(-50%);
+      height: var(--dsc-search-icon-size);
+      aspect-ratio: 1 / 1;
+      mask: var(--dsc-search-input-icon) center / contain no-repeat;
+      background: currentcolor;
+      z-index: 2;
+      pointer-events: none;
+    }
+  }
+  & input {
+    grid-column: 1 / 2;
+    grid-row: 1 / 2;
+
+    &:focus-visible {
+      z-index: 1;
+    }
+    &::-webkit-search-decoration,
+    &::-webkit-search-cancel-button {
+      appearance: none;
+    }
+  }
 
-.ds-search--lg .ds-search__input {
-  --dsc-search-clear-button-size: var(--ds-sizing-12);
+  &:has(input:placeholder-shown) [type='reset'] {
+    display: none;
+  }
 
-  height: var(--ds-sizing-14);
-  padding: 0 var(--ds-spacing-5);
-  padding-right: 2em;
-}
+  & [type='reset'] {
+    --dsc-button-padding: 0;
+
+    grid-column: 1 / 2;
+    grid-row: 1 / 2;
+    justify-self: end;
+    align-self: center;
+    margin-right: var(--ds-spacing-3);
+    border: none;
+    cursor: pointer;
+    height: var(--dsc-search-clear-button-size);
+    width: var(--dsc-search-clear-button-size);
+    border-radius: var(--ds-border-radius-default);
+    font-size: 0.5em;
+    z-index: 2;
+    position: relative;
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: 50%;
+      top: 50%;
+      transform: translate(-50%, -50%);
+      height: var(--dsc-search-clear-icon-size);
+      aspect-ratio: 1 / 1;
+      mask: var(--dsc-search-clear-icon) center / contain no-repeat;
+      background: currentcolor;
+      z-index: 2;
+      pointer-events: none;
+    }
+  }
 
-.ds-search--lg .ds-search__icon {
-  left: var(--ds-spacing-5);
-}
+  & [type='submit'] {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
 
-.ds-search__input.ds-search__input--simple {
-  padding-left: 2.4em;
-}
+    &:focus-visible {
+      z-index: 1;
```

Comment:
Comment about what this does? ☺️ 
Or maybe we can even do:
```
.ds-search {
  &:focus-visible { z-index: 1 }
}
```
...to catch both input and button? :)

Reply:
good catch